lazer/contracts/solana/programs/pyth-lazer-solana-contract/src/lib.rs

```diff
@@ -116,6 +168,18 @@ pub struct Initialize<'info> {
         bump,
     )]
     pub storage: Account<'info, Storage>,
+    #[account(
+        init,
+        payer = payer,
+        space = 0,
+        owner = system_program::ID,
+        seeds = [TREASURY_SEED],
+        bump,
+    )]
+    /// CHECK: this is a system program account but using anchor's `SystemAccount`
+    /// results in invalid output from the Accounts proc macro. No extra checks
+    /// are necessary because all necessary constraints are specified in the attribute.
+    pub treasury: AccountInfo<'info>,
```

Comment:
Can you tell me why treasury is owned by system_program? If it's owned by system program then we can't withdraw.

Reply:
I think we should just make it an address, not a PDA. Because if it's a PDA then we need a permissioned instruction to withdraw the fees. However with a simple address we can use the DAO treasury or what ever we want.

Reply:
Hmmm not a bad idea. if you make it an address we need to put it in the config then (and people need to look it up) which is a slightly worse ux.

Reply:
We need to save it in a PDA so that we can verify that it's the correct address to transfer funds to. Our SDK can load it by making an RPC call.

Reply:
System program's transfer is only possible if the account is owned by a system program.

I've added the configurable treasury address and a migration instruction and test.

Reply:
well in this case what you can do is manually += -= lamports from the accounts because both are writable.